src/zenml/cli/hub.py

```diff
@@ -0,0 +1,1203 @@
+    plugin_name: str,
+    version: Optional[str] = None,
+    output_dir: Optional[str] = None,
+) -> None:
+    """Clone a plugin from the hub.
+
+    Args:
+        plugin_name: Name of the plugin.
+        version: Version of the plugin. If not specified, the latest version
+            will be used.
+        output_dir: Output directory to clone the plugin to. If not specified,
+            the plugin will be cloned to a directory with the same name as the
+            plugin in the current working directory.
+    """
+    with event_handler(
+        event=AnalyticsEvent.ZENML_HUB_PLUGIN_CLONE,
+        metadata={
+            "plugin_name": plugin_name,
+            "plugin_version": version,
+        },
+    ) as analytics_handler:
+        client = HubClient()
+        analytics_handler.metadata["hub_url"] = client.url
+        plugin_name, author = _parse_plugin_name(plugin_name)
+        analytics_handler.metadata["plugin_name"] = plugin_name
+        analytics_handler.metadata["plugin_author"] = author
+        display_name = _plugin_display_name(plugin_name, version, author)
+
+        # Get plugin from hub
+        plugin = client.get_plugin(
+            plugin_name=plugin_name, plugin_version=version, author=author
+        )
+        if not plugin:
+            error(f"Could not find plugin '{display_name}' on the hub.")
+        analytics_handler.metadata["plugin_version"] = plugin.version
+
+        repo_url = plugin.repository_url
+        subdir = plugin.repository_subdirectory
+        commit = plugin.repository_commit
+
+        # Clone the source repo
+        if output_dir is None:
+            output_dir = os.path.join(os.getcwd(), plugin_name)
+        logger.info(f"Cloning plugin '{display_name}' to {output_dir}...")
+        # If no subdir is set, we can clone directly into output_dir
+        if not subdir:
+            repo_path = plugin_dir = output_dir
+        # Otherwise, we need to clone into a random dir and then move the subdir
+        else:
+            random_repo_name = f"_{uuid4()}"
+            repo_path = os.path.abspath(
+                os.path.join(os.getcwd(), random_repo_name)
+            )
+            plugin_dir = os.path.join(repo_path, subdir)
+
+        # Clone the repo and move it to `output_dir`
+        try:
+            _clone_repo(url=repo_url, to_path=repo_path, commit=commit)
+            shutil.move(plugin_dir, output_dir)
+            logger.info(f"Successfully Cloned plugin '{display_name}'.")
+        except GitCommandError:
+            error(
+                f"Could not find commit '{commit}' in repository '{repo_url}' "
+                f"of plugin '{display_name}'. This might happen if the owner "
+                "of the plugin has force-pushed to the plugin repository. "
+                "Please report this plugin version in the ZenML Hub or via "
+                "Slack."
+            )
+        finally:
+            shutil.rmtree(repo_path, ignore_errors=True)
```

Comment:
If `plugin.subdirectory` is `None`, this will delete the just-cloned repo right?

Reply:
It should always delete the repo, the idea here is we:
- clone the repo to a temp dir
- move the plugin dir (either repo_root or repo_root/subdir) to the output_dir
- delete the temp dir

Reply:
But if `plugin.subdirectory` is `None`, this will also delete the `output_dir` right? Which I'm assuming will be used downstream?

A little above:
```python
if not subdir:
  repo_path = plugin_dir = output_dir
```

Reply:
ah yes, I missed the part you highlighted. Should also be fixed in the new version now.